scipy/stats/_continuous_distns.py

```diff
@@ -5439,18 +5439,21 @@ def f3(q, h, k):
                            default=np.nan)
 
     def _stats(self, h, k):
-        if h >= 0 and k >= 0:
-            maxr = 5
-        elif h < 0 and k >= 0:
-            maxr = int(-1.0/h*k)
-        elif k < 0:
-            maxr = int(-1.0/k)
-        else:
-            maxr = 5
-
-        outputs = [None if r < maxr else np.nan for r in range(1, 5)]
+        maxr = np.where(np.logical_and(h >= 0, k >= 0), 5, np.nan)
+        maxr = np.where(np.logical_and(h < 0, k >= 0, np.isnan(maxr)),
+                        (-1.0/h*k).astype(int), maxr)
+        maxr = np.where(np.logical_and(k < 0, np.isnan(maxr)),
+                        (-1.0/k).astype(int), maxr)
+        maxr = np.where(np.isnan(maxr), 5, maxr)
```

Comment:
Consider simplifying:
```python3
        maxr = np.where((h < 0) & (k >= 0), (-1.0/h*k).astype(int), 5)
        maxr = np.where(k < 0, (-1.0/k).astype(int), maxr)
```
Also, the preference in the rest of stats distributions would be to rewrite this with `_lazyselect` or `_lazywhere`. See the rest of `kappa4`'s implementation.